csharp/src/Ice/OutgoingRequestFrame.cs

```diff
@@ -0,0 +1,131 @@
+        /// <param name="operation">The operation to invoke on the target Ice object.</param>
+        /// <param name="idempotent">True when operation is idempotent, otherwise false.</param>
+        /// <param name="context">An optional explicit context. When non null, it overrides both the context of the
+        /// proxy and the communicator's current context (if any).</param>
+        /// <param name="format">The Slice format (Compact or Sliced) used by the encapsulation.</param>
+        /// <param name="payloadWriter">An action that writes the contents of the payload.</param>
+        public OutgoingRequestFrame(IObjectPrx proxy, string operation, bool idempotent, Context? context,
+                                    FormatType? format, Action<OutputStream> payloadWriter)
+            : this(proxy.Communicator, proxy.Identity, proxy.Facet, operation, idempotent, proxy.Context, context)
+        {
+            StartEncapsulation(proxy.EncodingVersion, format);
+            payloadWriter(this);
+            EndEncapsulation();
+        }
+
+        /// <summary>Creates a new outgoing request frame with a null format.</summary>
+        /// <param name="proxy">A proxy to the target Ice object. This method uses the communicator, identity, facet,
+        /// encoding and context of this proxy to create the request frame.</param>
+        /// <param name="operation">The operation to invoke on the target Ice object.</param>
+        /// <param name="idempotent">True when operation is idempotent, otherwise false.</param>
+        /// <param name="context">An optional explicit context. When non null, it overrides both the context of the
+        /// proxy and the communicator's current context (if any).</param>
+        /// <param name="payloadWriter">An action that writes the contents of the payload.</param>
+        public OutgoingRequestFrame(IObjectPrx proxy, string operation, bool idempotent, Context? context,
+                                    Action<OutputStream> payloadWriter)
+            : this(proxy, operation, idempotent, context, format:null, payloadWriter)
+        {
+        }
+
+        /// <summary>Creates a new outgoing request frame with the given payload.</summary>
+        /// <param name="proxy">A proxy to the target Ice object. This method uses the communicator, identity, facet
+        /// and context of this proxy to create the request frame.</param>
+        /// <param name="operation">The operation to invoke on the target Ice object.</param>
+        /// <param name="idempotent">True when operation is idempotent, otherwise false.</param>
+        /// <param name="context">An optional explicit context. When non null, it overrides both the context of the
+        /// proxy and the communicator's current context (if any).</param>
+         /// <param name="payload">The payload of this request frame, which represents the marshaled in-parameters.
+        /// </param>
```

Comment:
Should this be on the line above?

Reply:
No, it would exceed 120 characters. There was however an extra space before the ///.